OctoKit/NotificationThread.swift

```diff
@@ -0,0 +1,336 @@
+    case markNotificationsRead(Configuration,String,Bool)
+    case getNotificationThread(Configuration,String)
+    case markNotificationThreadAsRead(Configuration,String)
+    case getThreadSubscription(Configuration,String)
+    case setThreadSubscription(Configuration,String,Bool)
+    case deleteThreadSubscription(Configuration,String)
+    case listRepositoryNotifications(Configuration,String,String,Bool,Bool,String?,String?,String,String)
+    case markRepositoryNotificationsRead(Configuration,String,String,String?)
+    
+    var configuration: Configuration {
+        switch self {
+            case .readNotifications(let config,_,_,_,_): return config
+            case .markNotificationsRead(let config, _, _): return config
+            case .getNotificationThread(let config, _),.markNotificationThreadAsRead(let config, _): return config
+            case .getThreadSubscription(let config, _): return config
+            case .setThreadSubscription(let config, _, _): return config
+            case .deleteThreadSubscription(let config, _): return config
+            case .listRepositoryNotifications(let config, _, _, _, _, _, _, _, _): return config
+            case .markRepositoryNotificationsRead(let config, _, _, _): return config
+        }
+    }
+    
+    var method: HTTPMethod {
+        switch self {
+            case .readNotifications(_, _, _, _, _),
+                 .getNotificationThread(_, _),
+                 .getThreadSubscription(_, _),
+                 .listRepositoryNotifications(_, _, _, _, _, _, _, _, _):
+                return .GET
+            case .markNotificationsRead(_, _, _),
+                 .setThreadSubscription(_, _, _),
+                 .markRepositoryNotificationsRead(_, _, _, _):
+                return .PUT
+            case .markNotificationThreadAsRead(_, _):
+                return .POST
+            case .deleteThreadSubscription(_, _):
+                return .DELETE
+        }
+    }
+    
+    var encoding: HTTPEncoding {
+        return .url
+    }
+    
+    var path: String {
+        switch self {
+            case .readNotifications(_, _, _, _, _),
+                 .markNotificationsRead(_, _, _):
+                return "notifications"
+            case .getNotificationThread(_, let threadID):
+                return "notifications/threads/\(threadID)"
+            case .markNotificationThreadAsRead(_, _):
+                return "notifications/threads/"
+            case .getThreadSubscription(_, let threadId),
+                 .setThreadSubscription(_, let threadId, _),
+                 .deleteThreadSubscription(_, let threadId):
+                return "notifications/threads/\(threadId)/subscription"
+            case .listRepositoryNotifications(_, let owner, let repo, _, _, _, _, _, _),
+                 .markRepositoryNotificationsRead(_, let owner, let repo, _):
+                return "/repos/\(owner)/\(repo)/notifications"
```

Comment:
Please remove the leading forward slash as this removes any path coming before this, e.g. if someone hosts their GitHub Enterprise instance on `https://example.com/scm/` calling this route would go to `https://example.com/repos…` instead of `https://example.com/scm/repos/`